pkg/logging/http.go

```diff
@@ -0,0 +1,55 @@
+package logging
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+type tracingResponseWriter struct {
+	next       http.ResponseWriter
+	statusCode int
+}
+
+func (w *tracingResponseWriter) Header() http.Header {
+	return w.next.Header()
+}
+
+func (w *tracingResponseWriter) Write(data []byte) (int, error) {
+	// While this looks slightly ugly, it follows the specification
+	// for an http.ResponseWriter according to:
+	//   https://golang.org/pkg/net/http/#ResponseWriter
+	if w.statusCode == 0 {
```

Comment:
This doesn't get handled internally?

Reply:
Nope, it is handled by `next`, but that does not propagate back up again.